db/user.go

```diff
@@ -221,3 +222,45 @@ func SyncUsers(course *model.Course, newUsers map[string]*model.User,
 
     return syncResult, nil;
 }
+
+// Insert method that takes a course, email [] string and resolves based on *, roles, emails to all emails
+func ResolveUsers(course *model.Course, emails []string) ([]string, error) {
+    if (backend == nil) {
+        return nil, fmt.Errorf("Database has not been opened.");
+    }
+    emailSet := map[string]any{};
+    roleSet := map[model.UserRole]any{};
+    // check for emails, roles, and * (all)
+    for _, email := range emails {
+        if (strings.Contains(email, "@") || (email == "")) {
+            emailSet[strings.ToLower(email)] = nil;
+        } else {
+            if (email == "*") {
+                allRoles := model.GetAllRoles();
+                for role := range allRoles {
+                    roleSet[role] = nil;
+                }
+            } else {
+                roleSet[model.GetRole(strings.ToLower(email))] = nil;
```

Comment:
What about invalid role strings?
Seems like a warning should be issued on those.